app/main.py

```diff
@@ -63,22 +62,27 @@ def clear(self) -> None:
         self.hash_table = [None] * self.capacity
         self.length = 0
 
-    def pop(self, key: Any) -> None:
-        try:
-            self.pop(key)
-        except KeyError:
-            return
+    def pop(self, key: Hashable, default: None = None) -> None:
+        if self[key]:
+            value = self[key]
+            del self[key]
+            return value
+        else:
```

Comment:
do you really need this `else` statement after `return`? It will never go to this line if statement in line66 is True